### PR TITLE
Create hooks-warnings.yml

### PR DESCRIPTION
**Pull request by ericLemanissier**
This hooks processes the pull request comments, in order to find build logs posted by @conan-center-bot, and extract all the conan-center hooks warnings, and reports them (after deduplication) as a comment on the PR.
You can play with it by copy/pasting any message from @conan-center-bot on this [PR](https://github.com/ericLemanissier/conan-center-index/pull/15) and see what the @github-actions bot posts afterwards.

In the current state of this PR, the bot does not check for the author of the comment, and it checks all the links having text `All logs` and url starting with `https://c3i.jfrog.io/c3i/misc/summary.html?json=`

One of the limitations is that @conan-center-bot does not always provide a link to the build log (eg if the recipe was not modified since the last build)

All suggestions are welcome

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: .github/workflows/hooks-warnings.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  comment:
+    if: ${{ github.event.issue.pull_request }}
```
Comment (prince-chrismc) [inline .github/workflows/hooks-warnings.yml:7]: Should the job also be limited on the comment author? Should this run in the comment is not from @conan-center-bot ?

File: .github/workflows/hooks-warnings.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+              for (const [ref, warnings] of warnings_map)
+              {
+                comment += "<details>\n"
+                comment += "<summary>" + ref + "</summary>\n\n```\n"
+                for(const warning of warnings)
+                  comment += warning + "\n"
+                comment += "\n```\n</details>\n"
+              }
+              comment += "</details>\n"
+              github.rest.issues.createComment({
```
Comment (prince-chrismc) [inline .github/workflows/hooks-warnings.yml:59]: Will the token from the public forks have enough permissions to post comments? I usually assume not, but perhaps there's trick here?

File: .github/workflows/hooks-warnings.yml (hunk 1 above)
Comment (ericLemanissier) [reply .github/workflows/hooks-warnings.yml:7]: Yes, I'm not doing it currently,  in order to be able to do some manual tests

File: .github/workflows/hooks-warnings.yml (hunk 2 above)
Comment (ericLemanissier) [reply .github/workflows/hooks-warnings.yml:59]: According to https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token it depends on the enterprise/organization/repository setting

**Comment by uilianries** [comment]
How did you test it?

**Comment by ericLemanissier** [comment]
as I mentioned in the PR message, by copy-pasting conan-center-bot message into this pr: https://github.com/ericLemanissier/conan-center-index/pull/15 (you are encouraged to try it!)

**Comment by czoido** [comment]
Very nice idea 👍

File: .github/workflows/hooks-warnings.yml (hunk 2 above)
Comment (jgsogo) [reply .github/workflows/hooks-warnings.yml:59]: I think it should work. The action triggers on comments, so it is somehow internal to the repository, the `GITHUB_TOKEN` should have default permissions (write). Anyway, usually the best way to test it is just to run it.

**Comment by jgsogo** [comment]
A couple of kindly requests:
 * limit the execution to comments by @conan-center-bot in pull-requests (I know it has been considered above, but one more time so we don't forget about it)
 * can you please parse and include the commit (at least a link to the comment) it has used? <-- Note to myself: probably we should add some metainfo (not visible) to these comments

**Comment by ericLemanissier** [comment]
> A couple of kindly requests:
>  * limit the execution to comments by @conan-center-bot in pull-requests (I know it has been considered above, but one more time so we don't forget about it)
Yes, I'll do it
>  * can you please parse and include the commit (at least a link to the comment) it has used? <-- Note to myself: probably we should add some metainfo (not visible) to these comments
Can the commit be extracted from the logs?

**Comment by jgsogo** [comment]
> Can the commit be extracted from the logs?

No. The logs contains the information from `conan install...`, `conan test...`, they don't contain the actual logs from Jenkins, which are the ones that would contain that commit. 

At the moment it can only be read from the comment message itself:
image
...but I think we should add some hidden metadata content in the message, like the following, something that could be easily parseable:

```
<!-- 
C3i meta
---
pr-number: xxx
author: xxxx
head-commit: xxxx
commit: xxxx
...
library: yyyy
versions:
 - version: x.y.z
 - result: yyyy
 
 -->
```

**Review by SSE4 — APPROVED** [review]

**Comment by ericLemanissier** [comment]
ok, so it now processes only messages from conan-center-bot, and shows the commit to from which it comes from

**Comment by ericLemanissier** [comment]
Friendly ping @SSE4 @jgsogo can you please re-review?

**Review by SSE4 — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@prince-chrismc @uilianries @czoido @jgsogo thanks for your feedback. If you still have some time available for a review, it'd be much appreciated !

**Comment by jgsogo** [comment]
https://github.com/conan-io/conan-center-index/pull/9655#issuecomment-1130799713

**Comment by uilianries** [comment]
Amazing! 👏

**Comment by SpaceIm** [comment]
Each time I write a comment in a PR of conan-center-index, I receive a github notification from this hook.

**Comment by ericLemanissier** [comment]
@SpaceIm this is surprising. What kind of notification ?

**Comment by SpaceIm** [comment]
https://github.com/conan-io/conan-center-index/actions/runs/2353341067

**Comment by ericLemanissier** [comment]
Mmmm, does it go away if you tick "Send notifications for failed workflows only" in https://github.com/settings/notifications ?

**Comment by jgsogo** [comment]
> Each time I write a comment in a PR of conan-center-index, I receive a github notification from this hook.

OMG! There were a few notifications... 

I've checked and I cannot see anything on this repo configuration/settings. From the documentation (https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/notifications-for-workflow-runs) I understand that it should notify you if it were a _scheduled job_, but this is not the case.

Maybe you can ask in Github forums, mention us, so we know if there is anything we can do on our side. I understand you want to keep receiving notifications on other repositories and disable this one.

**Pull request merged**